### PR TITLE
[RA1 Ch03] Align with RM 2.4

### LINKED ISSUE #2443: [RA1 Ch03] Align with RM 2.4 changes
Align 3.2.2 and 3.3.4 with RM 2.4

delete reference to RM 4.2

Consistent flavour spelling except in OpenStack terms

### PR DESCRIPTION
**Pull request by pgoyal01**
Fixes #2443

### CODE REVIEW

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 1]
```diff
@@ -79,7 +80,7 @@ This RA does not intend to restrict how workloads are distributed across tenants
 ### 3.2.2. Virtual Compute (vCPU and vRAM)
 The virtual compute resources (vCPU and vRAM) used by the VNFs behave like their physical counterparts.  A physical core is an actual processor and can support multiple vCPUs through Simultaneous Multithreading (SMT) and CPU overbooking. With no overbooking and SMT of 2 (2 threads per core), each core can support 2 vCPUs. With the same SMT of 2 and overbooking factor of 4, each core can support 8 vCPUs. The performance of a vCPU can be affected by various configurations such as CPU pinning, NUMA alignment, and SMT.
 
-The configuration of the virtual resources will depend on the profile and the flavour needed to host VNF components. Profiles are defined in the chapters 5.1 and 5.2 of the reference model document. Flavors are defined in the chapter 4.2 of the reference model document.
+The configuration of the virtual resources will depend on the software and hardware profiles and the flavour (resource sizing) needed to host VNF components. Profiles are defined in the [Reference Model chapter 5](../../ref_model/chapters/chapter05.md).
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter03.md:83]:
```suggestion
The configuration of the virtual resources will depend on the software and hardware profiles and the flavour (resource sizing) needed to host VNF components. Profiles are defined in the [Reference Model chapter 5](../../../ref_model/chapters/chapter05.md).
```

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 2]
```diff
@@ -254,7 +254,7 @@ An over use of Host Aggregates and Availability Zones can result in a granular p
 
 <a name="3.3.4"></a>
 ### 3.3.4. Flavor management
-A flavor defines the compute, memory, and storage capacity of nova instances. When instances are spawned, they are mapped to flavors which define the available hardware configuration for them. For simplicity, the flavors can be named as described in RM  like .tiny, .small, .medium, .large, .2xlarge and so on. The specifications for these sizes should map to the predefined compute flavors listed [here](../../../ref_model/chapters/chapter04.md#predefined-compute-flavours).
+In OpenStack a flavor defines the compute, memory, and storage capacity of nova instances. When instances are spawned, they are mapped to flavors which define the available hardware configuration for them. For simplicity, operators may create named flavors specifying both the sizing and the [software and hardware profile configurations](../../ref_model/chapters/chapter05.md).
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter03.md:257]:
```suggestion
In OpenStack a flavor defines the compute, memory, and storage capacity of nova instances. When instances are spawned, they are mapped to flavors which define the available hardware configuration for them. For simplicity, operators may create named flavors specifying both the sizing and the [software and hardware profile configurations](../../../ref_model/chapters/chapter05.md).
```

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 3]
```diff
@@ -293,10 +293,10 @@ The aim is to specify the requirements on deploying the VIM, from ground up (in
 Cloud Infrastructure physical Nodes
 
 The physical resources required for the Cloud Infrastructure are mainly based on COTS X86 hardware for control and data plane nodes.
-HW profiles are defined in the chapters 5.3 and 5.4 of the reference model document.
+HW profiles are defined in [Reference Model chapters 5.3 and 5.4](../../ref_model/chapters/chapter05.md).
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter03.md:296]:
```suggestion
HW profiles are defined in [Reference Model chapters 5.3 and 5.4](../../../ref_model/chapters/chapter05.md).
```

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 4]
```diff
@@ -49,14 +49,15 @@ This chapter is organized as follows:
...
-    - Flavor management
+    - Flavor* management
 *	Underlying Resources: are what provides the resources that allow the Consumable Infrastructure Resources and Services to be created and managed by the Cloud Infrastructure Management Software (VIM).
     - Virtualisation
     - Physical infrastructure
       -	Compute
       -	Network: Spine/Leaf; East/West and North/South traffic
       -	Storage
 
+> * Please note "flavours" is used in the Reference Model and shall continue to be used in the context of specify the geometry of the virtual resources. The term "flavor" will be used in OpenStack context including when specifying configurations; the OpenSTack term flavor includes what profile configuration information as "extra specs".
```
Comment (karinesevilla) [inline doc/ref_arch/openstack/chapters/chapter03.md:60]: Typo: OpenSTack
Is "what" necessary in the last sentence?

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter03.md]: @karinesevilla Corrected that and other errors - Thanks

**Review by karinesevilla — APPROVED** [review]

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 5]
```diff
@@ -49,14 +49,15 @@ This chapter is organized as follows:
...
-    - Flavor management
+    - Flavor* management
 *	Underlying Resources: are what provides the resources that allow the Consumable Infrastructure Resources and Services to be created and managed by the Cloud Infrastructure Management Software (VIM).
     - Virtualisation
     - Physical infrastructure
       -	Compute
       -	Network: Spine/Leaf; East/West and North/South traffic
       -	Storage
 
+> * Please note "flavours" is used in the Reference Model and shall continue to be used in the context of specifing the geometry of the virtual resources. The term "flavor" will be used in the OpenStack and this document context including when specifying configurations; the OpenStack term flavor includes the profile configuration information as "extra specs".
```
Comment (sukhdevkapur) [inline doc/ref_arch/openstack/chapters/chapter03.md:60]: Spelling - should be flavors

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter03.md:60]: @sukhdevkapur RM uses "flavours" for sizing and profiles for "host configurations". OpenStack, as you know, doesn't have host configurations but uses flavors for both sizing and in "extra-specs" to specify the host configuration desired by workloads.

RA1 refers to both RM and OSTK and the Note just points out these term differences.

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter03.md:60]: @sukhdevkapur There are now only two occurrences of "flavour" - 1. In the Note as to the difference between RM flavour and RA1 OSTK flavor, and another where there is again a reference to RM. 

We could delete the Note and edit the second use such that it doesn't have to use "flavour".

**Review by walterkozlowski — APPROVED** [review]

**Comment by pgoyal01** [comment]
@sukhdevkapur, @joaoarodrigues  Changed document to address your concerns. Please review. Tomorrow is last date. Thanks

File: doc/ref_arch/openstack/chapters/chapter03.md (hunk 5 above)
Comment (joaoarodrigues) [reply doc/ref_arch/openstack/chapters/chapter03.md:60]: @pgoyal01 Could you please indicate from where "flavours" is coming from - the source? (it is being created as a "surrogate" of OpenStack Flavors, so its specific to Anuket)

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter03.md:60]: @joaoarodrigues No "flavours" are not a surrogate for OpenStack flavors. The note states clearly that "flavours" is a term used in the RM. The note also makes clear that OpenStack flavor is also used to specify configuration and not only sizing; in RM configurations are specified by "profiles."

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter03.md:60]: In previous versions of RM, we had a table that specified various flavours -- tiny, small, large, etc. - and the vCPU, RAM, disk, etc. sizing. The flavours were used with the profiles (such as Basic). We have deleted the pre-defined flavours from RM and that may be the cause of some of the confusion.

**Review by collivier — APPROVED** [review]

**Comment by collivier** [comment]
All comments seem resolved

**Pull request merged**